Aquarius/Src/Aquarius/Renderer/Texture.cpp

```diff
@@ -0,0 +1,59 @@
+#include "Texture.h"
+
+
+namespace Aquarius {
+
+	uint8_t Texture::s_NextTextureUnitNum = 0;
+
+	Texture::Texture(const std::string& texture, TextureConfiguration configuration, bool hasAlpha)
+		: m_Configuration(configuration)
+	{
+		generateTexture(texture, hasAlpha);
+	}
+
+	bool Texture::generateTexture(const std::string& texture, bool hasAlpha)
+	{
+		bool success = false;
+
+		glGenTextures(1, &m_ID);
+		glBindTexture(GL_TEXTURE_2D, m_ID);
+		m_TextureUnitNum = GL_TEXTURE0 + s_NextTextureUnitNum++;
+
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, static_cast<int>(m_Configuration.textureWrapS));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, static_cast<int>(m_Configuration.textureWrapT));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, static_cast<int>(m_Configuration.minTextureFiltering));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, static_cast<int>(m_Configuration.magTextureFiltering));
+		unsigned char* data = stbi_load(texture.c_str(), &m_Width, &m_Height, &m_NumColorChannels, 0);
+		
+		if (data != nullptr)
+		{
+			int format = hasAlpha ? GL_RGBA : GL_RGB;
+
```

Comment:
Wonder if its better to use GLenum here, or even uint32_t (probably not a big deal for engine code to use GL stuff)

Reply:
To avoid narrowing conversions, GLenum is 32 bits

Reply:
Good call